rclrs/build.rs

```diff
@@ -27,10 +28,68 @@ fn main() {
     let ament_prefix_var_name = "AMENT_PREFIX_PATH";
     let ament_prefix_var = env::var(ament_prefix_var_name);
 
+    // # ALGORITHM #
+    // Search through ament index at AMENT_PREFIX_PATH../share/ament_index/resource_index/packages/ to find packages to include
+    // Packages will be located at (for each path in AMENT_PREFIX_PATHS):
+    // > AMENT_PREFIX_PATH/include/ (For the header files) (OLD STYLE)
+    //      -isystemAMENT_PREFIX_PATH/include
+    // > AMENT_PREFIX_PATH/include/project_name (NEW STYLE)
+    //      -isystemAMENT_PREFIX_PATH/include/project_name (if and only if it has a folder inside of it also named project_name)
+    //      -isystemAMENT_PREFIX_PATH/include/CycloneDDS (special case, match for this)
+    // -OR-
+    // > AMENT_PREFIX_PATH/lib  (For the link libraries)
+    //
+    // See REP 122 for more details: https://www.ros.org/reps/rep-0122.html#filesystem-layout
+
     if let Ok(ament_prefix_paths) = ament_prefix_var {
-        for ament_prefix_path in ament_prefix_paths.split(':') {
-            builder = builder.clang_arg(format!("-I{}/include", ament_prefix_path));
-            println!("cargo:rustc-link-search=native={}/lib", ament_prefix_path);
+        for ament_prefix_path in ament_prefix_paths.split(':').filter_map(|p| {
+            let path = Path::new(p).to_path_buf();
+            if path.is_dir() {
```

Comment:
This check is redundant with the check further down – If the prefix doesn't exist, the ament index won't exist either. I'd remove it.